flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala

```diff
@@ -838,6 +842,311 @@ class TableSinkITCase extends StreamingTestBase {
 
   }
 
+  @Test
+  def testPartialInsert(): Unit = {
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE testSink (
+         |  `a` INT,
+         |  `b` DOUBLE
+         |)
+         |WITH (
+         |  'connector' = 'values',
+         |  'sink-insert-only' = 'false'
+         |)
+         |""".stripMargin)
+
+    val t = env.fromCollection(tupleData2).toTable(tEnv, 'x, 'y)
+    tEnv.createTemporaryView("MyTable", t)
+
+    tEnv.executeSql(
+      s"""
+         |INSERT INTO testSink (b)
+         |SELECT sum(y) FROM MyTable GROUP BY x
+         |""".stripMargin).await()
+    val expected = List(
+      "null,0.1",
+      "null,0.4",
+      "null,1.0",
+      "null,2.2",
+      "null,3.9")
+    val result = TestValuesTableFactory.getResults("testSink")
+    assertEquals(expected.sorted, result.sorted)
+  }
+
+  @Test
+  def testPartialInsertWithNotNullColumn(): Unit = {
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE testSink (
+         |  `a` INT NOT NULL,
+         |  `b` DOUBLE
+         |)
+         |WITH (
+         |  'connector' = 'values',
+         |  'sink-insert-only' = 'false'
+         |)
+         |""".stripMargin)
+
+    val t = env.fromCollection(tupleData2).toTable(tEnv, 'x, 'y)
+    tEnv.createTemporaryView("MyTable", t)
+
+    expectedEx.expect(classOf[ValidationException])
+    expectedEx.expectMessage("Column 'a' has no default value and does not allow NULLs")
```

Comment:
Flink does not support default value yet, we can improve exception message here, how about improve to 
```suggestion
    expectedEx.expectMessage("Column 'a' does not allow NULL value, thus it can not be used in partial insert")
```

Reply:
sounds good, will update soon.

Reply:
Check the error msg again and found the original one is more accuracy. So i will revert the change.